Add Duke Nukem 1, 2, and 3D - Atomic Edition

tomojira opened the pull request:
Added Duke Nukem 1 & 2, installed from the Duke Nukem 3D CD:

- [Duke Nukem 3D (USA)](http://redump.org/disc/2007/)

And added Duke Nukem 3D: Atomic Edition, installed from CD matching with Redump:

- [Duke Nukem 3D - Atomic Edition (USA)](http://redump.org/disc/20161/)

Atomic Edition doesn't include the first 2 games, hence the 2 different CDs.
I didn't include the TEN or Build Editor stuff during Duke Nukem 3D installation (they're still on the disc anyway - unfortunately, this huge CD image is needed to run the game).

The first 2 games have no config to worry about, they just work.
DOSC has auto start.

Like other Build Engine games, Duke Nukem 3D has issues with flickering graphics, so `cpu_year: 1996` is used to fix it.
DOSC has Sound Blaster 16 config, 640x480 resolution, and auto start.

televandalist commented:
The games were delisted from GOG, but I have them and could add the variants for them.

tomojira commented:
Updates:
- Changed to WAITMODECHANGE
- DN1 speed reduced with `cpu_year: 1985`
- "Duke Nukum" renamed to "Duke Nukem"
- DN3D audio track converted with CHDtoOGG
- DN3D changed to `cpu_max_year: 2000` - still fixes the flickering graphics, while allowing better performance in-game
- General MIDI option added for DN3D

tomojira commented:
More updates
- Consolidated Duke Nukem 1 with an episode menu (and hint sheet option)
- Added setup and help options to Duke Nukem 3D
- Added shareware variants for 1, 2, and 3D, just for the fun of sharing them
- Added the original 3-episode Duke Nukem 3D v1.3d. Maybe a bit overkill, but along the lines of what we've done with some expansions, could have nostalgic value for people who don't care about Atomic Edition and its 4th episode, and it also doesn't require a CD image
